### PR TITLE
Deprecate empty version string (0.8 backport)

### PR DESCRIPTION
**Pull request by omus**
Deprecating the functionality which will be added in:
- https://github.com/JuliaCloud/AWSS3.jl/pull/199

### CODE REVIEW

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
Reduced the scope of the backport

**Comment by omus** [comment]
bors try

File: test/awss3.jl [hunk 1]
```diff
@@ -195,6 +195,32 @@ function awss3_tests(config)
...
+        # Create the file to ensure we're only testing `version`
+        k = "version_empty.txt"
+        s3_put(config, bucket_name, k, "v1")
+        s3_put(config, bucket_name, k, "v2")
+
+        # Using an empty string as the version returns the latest version
+        @test s3_get(bucket_name, k; version="") == "v2"
+        @test s3_get_meta(bucket_name, k; version="") isa AbstractDict
+        @test s3_exists(bucket_name, k; version="")
+        @test s3_delete(bucket_name, k; version="") == UInt8[]
```
Comment (omus) [inline test/awss3.jl:208]: I should add depredations for these even though they will continue to work for MinIO

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
Failure looks unrelated to me.

**Comment by omus** [comment]
bors try

**Review by mattBrzezinski — APPROVED** [review]

**Comment by omus** [comment]
Now that #199 is merged. I'll try to merge this too. Hopefully bors works properly with a backport

**Pull request merged**